.github/actions/setup/action.yml

```diff
@@ -0,0 +1,127 @@
+name: Setup
+description: Setup the environment
+
+runs:
+  using: composite
+  steps:
+    - name: Install rust toolchain
+      uses: dtolnay/rust-toolchain@nightly
+
+    - name: Install cargo-nextest
+      uses: taiki-e/install-action@v2
+      with:
+        tool: cargo-nextest
+
+    - name: Install just
+      uses: extractions/setup-just@v1
+      with:
+        just-version: 1.5.0 # optional semver specification, otherwise latest
+
+    - name: Setup python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.10.11"
+        cache: "pip"
+
+    - name: Install python dependencies
+      run: pip install -r requirements.txt
+      shell: bash
+
+    - name: Enable node.js corepack
+      run: corepack enable
+      shell: bash
+
+    - name: Setup node.js
+      uses: actions/setup-node@v4
+      with:
+        cache: pnpm
+        node-version: "18"
+
+    - name: Install node.js dependencies
+      run: pnpm install
+      shell: bash
+
+    - name: (linux) install lavapipe, vulkan sdk, alsa
+      if: matrix.os == 'ubuntu-22.04'
```

Comment:
@FL33TW00D @ocavue this will not work, the `matrix` vars are not available in an action. The values for `matrix.os` and `matrix.platform` can be passed as action inputs: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs

Reply:
First, thanks for noticing this. I do agree that passing `os` and `platform` explicitly is a better option.

On the other hand, I believe using `if: matrix.os ==` in a composite workflow (i.e. `.github/actions/setup/action.yml`) does work. You can see the result from [this PR](https://github.com/ocavue/ratchet/pull/5), where I use `if: matrix.os ==` to print system information in different OSs, and it works as expected. 

Since FL33TW00D is doing some heavy work editing the CI workflow in another branch, I would like not to touch the CI config. Once his work is done, we can add action inputs as your described.

Reply:
@ocavue You are right. I did check [the run](https://github.com/huggingface/ratchet/actions/runs/9895313741/job/27334902256) and did not find the right step, my bad!
Apparently the [full context is passed to the action](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts), that's new to me.